assets/src/dashboard/app/font/fontProvider.js

```diff
@@ -57,15 +49,26 @@ function FontProvider({ children }) {
 
   useEffect(() => {
     let mounted = true;
-    getAllFonts().then((newFonts) => {
-      if (mounted) {
-        setFonts(newFonts);
-      }
-    });
+    async function loadFonts() {
+      const newFonts = await import(
+        /* webpackChunkName: "chunk-fonts" */ '../../../fonts/fonts'
+      );
+      const formattedFonts = newFonts.default.map((font) => ({
+        name: font.family,
+        value: font.family,
+        ...font,
+      }));
+
+      setFonts(formattedFonts);
+    }
+    if (mounted) {
+      loadFonts();
+    }
+
     return () => {
       mounted = false;
     };
-  }, [getAllFonts]);
+  }, [setFonts]);
```

Comment:
You don't need this dependency - `useState` update functions are guaranteed to be stable and `eslint` also won't complain if you remove it, as it knows that.

Reply:
```suggestion
  }, []);
```